cmd/assetsvc/handler.go

```diff
@@ -109,7 +109,12 @@ func getChartCategories(w http.ResponseWriter, req *http.Request, params Params)
 		return
 	}
 
-	chartCategories, err := getAllChartCategories(namespace, repo)
+	cq := ChartQuery{
+		namespace: namespace,
+		repos:     append(strings.Split(strings.TrimSpace(req.FormValue("repos")), ","), repo),
+	}
```

Comment:
Perhaps if you extract the above 15 lines into something like `extractChartQueryFromRequest()` it'd be easy to test in isolation? As it is, I think we need a test for the repos - it looks like you're creating a list by appending the current repo (from the url) with those from the submitted form data, but we may (?) want to ensure that is unique? Anyway, with that tested, +1

Reply:
Good idea indeed. I still think we should improve and normalize how we retrieve the params. It's such a mess being mixing up params from `req.FormValue("foo")` and from `params["foo"]`... but, yep, for this PR, I totally agree with you.

Regarding the uniqueness, it won't have any side effect if we pass dups since, in the end, it will become something like `WHERE ...OR repo=X OR repo=X`